core/app/models/spree/order.rb

```diff
@@ -75,11 +75,19 @@ class CannotRebuildShipments < StandardError; end
     # Customer info
     belongs_to :user, class_name: Spree::UserClassHandle.new, optional: true
 
-    belongs_to :bill_address, foreign_key: :bill_address_id, class_name: 'Spree::Address', optional: true
+    belongs_to :bill_address,
+      foreign_key: :bill_address_id,
+      class_name: 'Spree::Address',
+      optional: true,
+      inverse_of: false
     alias_method :billing_address, :bill_address
     alias_method :billing_address=, :bill_address=
 
-    belongs_to :ship_address, foreign_key: :ship_address_id, class_name: 'Spree::Address', optional: true
+    belongs_to :ship_address,
+      foreign_key: :ship_address_id,
+      class_name: 'Spree::Address',
+      optional: true,
+      inverse_of: false
```

Comment:
Same. You did not added a `dependent: false` here, although mentioned in the comment.